components/concept/push/src/main/java/mozilla/components/concept/push/Push.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.concept.push
+
+/**
+ * The Push component that will handle push messages from the [PushService] provided.
+ * Starting Push in the Application's onCreate is recommended.
+ */
+class Push(
+    internal val service: PushService
+) {
+    /**
+     * Start the Push component.
+     */
+    fun init(): Push {
+        Push.instance = this
+
+        service.start()
+
+        return this
+    }
+
+    internal fun onNewToken(newToken: String) {}
```

Comment:
nit: We could refer to the issues number that this work is related, and the functions could  be equal to `Unit`until we provide an implementation.